### PR TITLE
Simplify runner code

### PR DESCRIPTION
**Pull request by dgageot**
Multiple small improvements to the runner code. More to come

### CODE REVIEW

File: pkg/skaffold/kubernetes/debugging/container_manager_test.go [hunk 1]
```diff
@@ -84,3 +84,11 @@ func TestContainerManager(t *testing.T) {
 		t.CheckDeepEqual(1, terminatedCount)
 	})
 }
+
+func TestContainerManagerZeroValue(t *testing.T) {
```
Comment (balopat) [inline pkg/skaffold/kubernetes/debugging/container_manager_test.go:87]: this is so cool that you can have nil dereference checks from within the instance methods

**Comment by balopat** [comment]
This is going to take some time to review - would you consider chopping it up to smaller PRs?

**Comment by dgageot** [comment]
@balopat let me split it

**Comment by tejal29** [comment]
@dgageot Can we close this PR if you are splitting this into small prs?

**Pull request closed without merging**